src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2720,7 +2721,30 @@ public static void TestNameWithInvariantCulture()
                 Assert.True(pacific.DisplayName.IndexOf("Pacific", StringComparison.OrdinalIgnoreCase) >= 0, $"'{pacific.DisplayName}' is not the expected display name for Pacific time zone");
 
             }).Dispose();
+        }
+
+        private static bool CanTestWindowsNlsDisplayNames => s_isWindows && RemoteExecutor.IsSupported && WindowsUILanguageHelper.GetInstalledWin32CulturesWithUniqueLanguages().Length > 1;
```

Comment:
>s_isWindows [](http://example.com/codeflow?start=61&length=11)

do we need this check if you already using [PlatformSpecific(TestPlatforms.Windows)]?

Reply:
I wasn't sure if the attributes were evaluated in any particular order by the test runners.  This seemed like a reasonable safety precaution.

Alternatively I could just have the helper method return empty if not on Windows.

Reply:
When having `[PlatformSpecific(TestPlatforms.Windows)]` it is guaranteed the test method will run on Windows platforms only. So, it doesn't matter when the attribute is evaluated.